jetty-util/src/main/java/org/eclipse/jetty/util/resource/JarFileResource.java

```diff
@@ -316,11 +317,12 @@ public long lastModified()
         }
 
         Enumeration<JarEntry> e = jarFile.entries();
-        String dir = _urlString.substring(_urlString.lastIndexOf("!/") + 2);
+        String encodedDir = _urlString.substring(_urlString.lastIndexOf("!/") + 2);
+        String dir = URIUtil.decodePath(encodedDir);
```

Comment:
Method `newConnection()` has a similar logic (searching for `!/`) and has been changed from using `_urlString` to use `_url.getPath()`. Why here we are not doing the same?

Reply:
Reverted change in `newConnection`, but with comment indicating assumption.